Bluejay/Bluejay/Bluejay.swift

```diff
@@ -380,9 +385,15 @@ public class Bluejay: NSObject {
      - Parameters:
         - peripheralIdentifier: The peripheral to connect to.
         - timeout: Specify how long the connection time out should be.
+        - options: Optional connection options, if not specified, Bluejay's default will be used.
         - completion: Called when the connection request has fully finished and indicates whether it was successful, cancelled, or failed.
     */
-    public func connect(_ peripheralIdentifier: PeripheralIdentifier, timeout: Timeout, completion: @escaping (ConnectionResult) -> Void) {
+    public func connect(
+        _ peripheralIdentifier: PeripheralIdentifier,
+        timeout: Timeout,
```

Comment:
Another question in the "function arguments vs. struct members" domain: Should "timeout" be in the ConnectionOptions struct a well? 

It does make more sense the is some ways, and it could be useful to be able to default it as well, but that would be a breaking API change, and it doesn't totally fit the behind the scenes meaning of that struct (mapping to the CB connection options dictionary), since timeout is a Bluejay feature.

As with using a options struct to do initial configuration mentioned above, my gut feeling is that change probably isn't worth it, but again wanted to just raise it for consideration.

Reply:
Good point, this makes me feel maybe `ConnectionOptions` is just poorly named. Yes that does map to CB's terminology, but since we're abstracting it, we could name it anything we want to further our goal of making it clearer and self-explanatory to the user. For example, `WarningOptions`. That way, the distinction between `Timeout` and `WarningOptions` is clear, and it's also clearer that if you want to work with `WarningOptions`, you're going to deal with some kind of warning behaviours related to connection. Would this be better?

Reply:
I think 'WarningOptions' probably WOULD be a better name for it as it stands right now. I

 guess whether to not we change it depends on how we want to evolve it in the future. If we think that there aren't going to be any new options added, or if they were added we would want to deal with them separately rather than wrapping them up in the same struct. If so, i'd say we rename to 'WarningOptions'. 

If we think there will be more stuff going in there in the future, or if we think we want to add the ability to default some other options that we add (like adding timeout to the struct so it becomes something you an set a default for), then I think ConnectionOptions is fine.

Reply:
After some more thinking and reviewing of all three CBConnectPeripheralOptions available, they are all really semantically closer to "warning options" than "connection options", so I don't really feel comfortable grouping those with timeout, which is definitely a connection option.

If we add something else similar to timeout in the future, then we can add a `ConnectionOptions` struct to group timeout with those new connection options, but even then, I would still prefer to separate them from CBConnectPeripheralOptions/WarningOptions.